pkg/handlers/archive.go

```diff
@@ -434,25 +436,26 @@ func (a *Archive) handleNestedFileMIME(ctx logContext.Context, tempEnv tempEnv,
 
 // determineMimeType reads from the provided reader to detect the MIME type.
 // It returns the detected MIME type and a new reader that includes the read portion.
-func determineMimeType(reader io.Reader) (mimeType, io.Reader, error) {
+func determineMimeType(reader io.Reader) (*mimetype.MIME, io.Reader, error) {
+	bufReader := bufio.NewReaderSize(reader, mimeTypeBufferSize)
```

Comment:
is creating a new reader so expensive that re-using this one makes sense? the downside of doing it this way is that now all readers have to use the mime-detection buffer size, which might not always be the best choice.